runtime/ops/os/mod.rs

```diff
@@ -186,6 +186,8 @@ fn op_get_exit_code(state: &mut OpState) -> i32 {
 
 #[op2(fast)]
 fn op_exit(state: &mut OpState) {
+  crate::ops::otel::otel_drop_state(state);
```

Comment:
We should have some tests that ensure the OTEL pipeline is properly flushed before the process exits. Some scenarios:

* `console.log("foo"); // Program naturally exits here`
* `console.log("foo"); Deno.exit(0);`
* `console.log("foo"); throw new Error("uncaught");`
* Web worker logs something  and immediate calls `self.close()`
* Web worker logs something and the main worker immediately calls `worker.terminate()`
* Web worker logs something and calls `Deno.exit()`
* Web worker logs something and throws an exception.